exporter/kafkaexporter/factory_test.go

```diff
@@ -139,36 +145,205 @@ func TestCreateLogsExporter_err(t *testing.T) {
+			}),
+		)
+
+		return cfg, f
+	}
+
+	traces := ptrace.NewTraces()
+	traces.ResourceSpans().AppendEmpty().ScopeSpans().AppendEmpty().Spans().AppendEmpty().SetName("testSpan")
+
+	t.Run("traces_custom_encoding", func(t *testing.T) {
+		cfg, exporterFactory := setupTest(customMarshalerEncoding, true)
+
+		exporter, err := exporterFactory.CreateTracesExporter(context.Background(), componenttest.NewNopExporterCreateSettings(), cfg)
+		consumeErr := exporter.ConsumeTraces(context.TODO(), traces)
+		require.NoError(t, err)
+		require.NoError(t, consumeErr)
+		require.NotNil(t, exporter)
+	})
+	t.Run("traces_default_encoding", func(t *testing.T) {
+		cfg, exporterFactory := setupTest(defaultEncoding, false)
+
+		exporter, err := exporterFactory.CreateTracesExporter(context.Background(), componenttest.NewNopExporterCreateSettings(), cfg)
+		consumeErr := exporter.ConsumeTraces(context.TODO(), traces)
+		require.NoError(t, err)
+		require.NoError(t, consumeErr)
+		assert.NotNil(t, exporter)
+	})
+
+	metrics := pmetric.NewMetrics()
+	metrics.ResourceMetrics().AppendEmpty().ScopeMetrics().AppendEmpty().Metrics().AppendEmpty().SetName("test")
+	t.Run("metrics_custom_encoding", func(t *testing.T) {
+		cfg, exporterFactory := setupTest(customMarshalerEncoding, true)
+
+		exporter, err := exporterFactory.CreateMetricsExporter(context.Background(), componenttest.NewNopExporterCreateSettings(), cfg)
+		consumeErr := exporter.ConsumeMetrics(context.TODO(), metrics)
+		require.NoError(t, err)
+		require.NoError(t, consumeErr)
+		require.NotNil(t, exporter)
+	})
+	t.Run("metrics_default_encoding", func(t *testing.T) {
+		cfg, exporterFactory := setupTest(defaultEncoding, false)
+
+		exporter, err := exporterFactory.CreateMetricsExporter(context.Background(), componenttest.NewNopExporterCreateSettings(), cfg)
+		consumeErr := exporter.ConsumeMetrics(context.TODO(), metrics)
+		require.NoError(t, err)
+		require.NoError(t, consumeErr)
+		assert.NotNil(t, exporter)
+	})
+
+	logs := plog.NewLogs()
+	logs.ResourceLogs().AppendEmpty().ScopeLogs().AppendEmpty().LogRecords().AppendEmpty().Body().SetStringVal("test")
+	t.Run("logs_custom_encoding", func(t *testing.T) {
+		cfg, exporterFactory := setupTest(customMarshalerEncoding, true)
+
+		exporter, err := exporterFactory.CreateLogsExporter(context.Background(), componenttest.NewNopExporterCreateSettings(), cfg)
+		consumeErr := exporter.ConsumeLogs(context.TODO(), logs)
+		require.NoError(t, err)
+		require.NoError(t, consumeErr)
+		require.NotNil(t, exporter)
+	})
+	t.Run("logs_default_encoding", func(t *testing.T) {
+		cfg, exporterFactory := setupTest(defaultEncoding, false)
+
+		exporter, err := exporterFactory.CreateLogsExporter(context.Background(), componenttest.NewNopExporterCreateSettings(), cfg)
+		consumeErr := exporter.ConsumeLogs(context.TODO(), logs)
+		require.NoError(t, err)
+		require.NoError(t, consumeErr)
+		assert.NotNil(t, exporter)
+	})
+
+}
+
+/*
```

Comment:
No commented out code to be committed, please ensure existing tests are correctly replaced.

Reply:
Will the checks run again?

Reply:
If it your first time contibuting to the project, it will need to be manually run by an approver.